apisix/balancer/roundrobin.lua

```diff
@@ -18,12 +18,19 @@
 local roundrobin  = require("resty.roundrobin")
 local core = require("apisix.core")
 local nkeys = core.table.nkeys
+local pairs = pairs
 
 
 local _M = {}
 
 
 function _M.new(up_nodes, upstream)
+    local safe_limit = 0
+    for _, weight in pairs(up_nodes) do
+        -- the weight can be zero
+        safe_limit = safe_limit + weight + 1
```

Comment:
I don't understand, why is `weight` needed here?
Is it better than `safe_limit = safe_limit + 1` ?

Reply:
The number of possible retry won't be larger than `weight + 1`.

Reply:
Thanks, got it.